### PR TITLE
DOMA-1356 mobile menu redesign

### PR DESCRIPTION
**Pull request by MrFoxPro**
image

### CODE REVIEW

**Comment by MrFoxPro** [comment]
image

File: apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx [hunk 1]
```diff
@@ -31,29 +38,21 @@ export const MobileUserMenu: React.FC = () => {
...
-                    <RestFilled style={menuIconStyles}/>
-                    {SignOutMessage}
-                </Space>
-            </StyledMenuItem>
-        </StyledMenu>
+    const [showModal, setShowModal] = useState(false)
+    const ModalView = () => (
+        <ModalWrapper>
+            <Button color={colors.white} onClick={auth.signout}>{ProfileMessage}</Button>
+            <Button color={colors.white} onClick={() => router.push('/user')}>{SignOutMessage}</Button>
```
Comment (sitozzz) [inline apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx:45]: May be wrap onClick content with useCallback hook?

File: apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx [hunk 2]
```diff
@@ -31,29 +38,21 @@ export const MobileUserMenu: React.FC = () => {
...
     return (
         auth.isAuthenticated
             ? (
-                <Dropdown overlay={DropdownOverlay} placement='bottomLeft'>
-                    <Button type={'inlineLink'} icon={<Avatar size={40} icon={<UserOutlined/>}/>}/>
-                </Dropdown>
+
+                <>
+                    <Button type={'inlineLink'} icon={<Avatar size={40} icon={<UserOutlined />} />} onClick={() => setShowModal(true)} />
+                    <Modal onCancel={() => setShowModal(false)} visible={showModal} modalRender={ModalView} style={modalStyle} />
```
Comment (sitozzz) [inline apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx:54]: Wrap with useCallback plz

Comment (MrFoxPro) [reply apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx:54]: not sure it is necessary, but I did it

**Review by sitozzz — APPROVED** [review]

File: apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx [hunk 3]
```diff
@@ -22,38 +22,49 @@ export const StyledMenu = styled(Menu)`
...
-                </Space>
-            </StyledMenuItem>
-            <StyledMenuItem key='signout' onClick={auth.signout}>
-                <Space size={16}>
-                    <RestFilled style={menuIconStyles}/>
-                    {SignOutMessage}
-                </Space>
-            </StyledMenuItem>
-        </StyledMenu>
+    const onSignOutClick = useCallback(() => router.push('/user'), [])
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx:41]: `auth.signout` instead `router.push('/user')` :)

File: apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx [hunk 4]
```diff
@@ -22,38 +22,49 @@ export const StyledMenu = styled(Menu)`
...
+
+    return (
+        <ModalWrapper>
+            <Button color={colors.white} onClick={auth.signout}>{ProfileMessage}</Button>
+            <Button color={colors.white} onClick={onSignOutClick}>{SignOutMessage}</Button>
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx:46]: swap `onClick` handlers

**Review by nomerdvadcatpyat — COMMENTED** [review]
Why can't I hide the menu by clicking or touching on the overlay?

изображение

File: apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx (hunk 4 above)
Comment (AntonAL) [reply apps/condo/domains/user/components/UserMenu/MobileUserMenu.tsx:46]: I've also noticed logging out on click on Profile button by manual testing ;)

**Review by AntonAL — COMMENTED** [review]
There is no way to close opened menu in mobile mode.
Maybe to add closing on click on overlay?

**Comment by MrFoxPro** [comment]
> There is no way to close opened menu in mobile mode. Maybe to add closing on click on overlay?

Done

**Review by nomerdvadcatpyat — APPROVED** [review]

**Pull request merged**